### DISCUSSION TITLE
Metrics endpoint not working

### DISCUSSION THREAD

**Discussion by ars3nicum** [Budibase/budibase#10249] [Help]
I am trying to get metrics according to [this instruction](https://docs.budibase.com/docs/metrics) and getting error:
`{"message":"Database name cannot be undefined.","status":500}`

Budibase version 2.5.0

What am I doing wrong?

**Comment by andz-bb** [comment]
hey @ars3nicum 

are you on cloud or a self hosted setup? If it's the latter, could you give some more details about your installation?

I've tested this on both cloud and self-hosted on v2.5.2 and the endpoint works ok on my instances. I believe this metrics endpoint is looking for your `global-db` on couch so could you check that it exists and is named properly? You can use [our docs on accessing your CouchDB here](https://docs.budibase.com/docs/accessing-couchdb) for reference.

**Comment by ars3nicum** [comment]
I updated budibase (self-hosted) to version 2.5.2 and endpoint is working now, but if I try to curl it with the api-key header I get the following:

```
curl --request GET \
  --url https:/<mybudi>/api/public/v1/metrics \
  --header 'x-budibase-api-key: <myapi>'
Warning: Binary output can mess up your terminal. Use "--output -" to tell
Warning: curl to output it to your terminal anyway, or consider "--output
Warning: <FILE>" to save to a file.


```

From prometheus I also get an error: `server returned HTTP status 403 Forbidden`

**Reply by andz-bb** [reply]
@ars3nicum glad to hear you're a step closer with it.

For the curl you need to add the `--compressed` flag to it so that it decompresses the response. I saw the same on my end with the curl example in the docs so I will update it now to reflect this.

I'm not sure if there's a way to include the `x-budibase-api-key` header out of the box on the prometheus scrape_configs so that's maybe why you are getting a 403 in this case. I'm not an expert on prometheus config so I can't help you further on this one, although it's likely someone else online has a workaround for this by using a reverse proxy or similar to include the header on the request.

**Comment by ars3nicum** [comment]
Thank you! Curl also works.
As for the Prometheus there is [example config](https://docs.budibase.com/docs/metrics#prometheus-scrape-config), but for me not working.

**Reply by odcheck** [reply]
I guess it ain't working cause the http basic auth part, this is a working snippet for the scrape job tha I use:

```
  - job_name: 'budibase'
    scrape_interval: 30s
    scheme: http
    metrics_path: /api/public/v1/metrics
    static_configs:
      - targets: ['IP-ADDRESS:10000']
    authorization:
      type: "x-budibase-api-key"
      credentials: "API-KEY"
 ```

Of course as you can see I missuse `type` so that I will be used as header
